### PR TITLE
docs(api): add hierarchical diagram

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch:
- kills the first level of table-of-contents (it's useless)
- adds hierarchical diagram to the api.md that briefly outlines puppeteer API structure.

Diagram (faded entites are not represented by pptr api right now): puppeteer overview

### CODE REVIEW

**Comment by pavelfeldman** [comment]
sgtm

File: docs/api.md [hunk 1]
```diff
@@ -6,180 +6,192 @@
...
 Puppeteer is a Node library which provides a high-level API to control Chromium over the DevTools Protocol.
 
+Puppeteer API is hierarchical and mirrors browser objects. On the following diagram, faded entities are not currently represented in Puppeteer.
+
+![puppeteer overview](https://user-images.githubusercontent.com/746130/31592143-089f6f9a-b1db-11e7-9a20-16b7fc754fa1.png)
+
+- [`Puppeteer`](#class-puppeteer) communicates with browser using [devtools protocol](https://chromedevtools.github.io/devtools-protocol/).
+- [`Browser`](#class-browser) instance owns multiple pages.
+- [`Page`](#class-page) has at least one frame: main frame. There might be other frames created by [iframe](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe) or [frame](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/frame) tags.
+- [`Frame`](#class-frame) has at least one execution context - default execution context - where frame's javascript is executed. Frame might have additional execution contexts associated with [extensions](https://developer.chrome.com/extensions).
```
Comment (ebidel) [inline docs/api.md:190]: JavaScript

File: docs/api.md [hunk 2]
```diff
@@ -6,180 +6,192 @@
...
+  * [response.url](#responseurl)
 
 <!-- tocstop -->
 
-## Puppeteer
+### Overview
 
 Puppeteer is a Node library which provides a high-level API to control Chromium over the DevTools Protocol.
 
+Puppeteer API is hierarchical and mirrors browser objects. On the following diagram, faded entities are not currently represented in Puppeteer.
```
Comment (ebidel) [inline docs/api.md:183]: browser objects? clarify

**Review by ebidel — APPROVED** [review]
Cool!

**Comment by aslushnikov** [comment]
@ebidel oops missed your comments! Will address in a follow-up

File: docs/api.md (hunk 1 above)
Comment (aslushnikov) [reply docs/api.md:190]: I'm *this close* to adding a lint check. Done.

File: docs/api.md (hunk 2 above)
Comment (aslushnikov) [reply docs/api.md:183]: @ebidel browser structure?

Comment (ebidel) [reply docs/api.md:183]: Ah. How about "mirrors a browser"

Comment (aslushnikov) [reply docs/api.md:183]: sg

**Pull request merged**